### PR TITLE
[skrifa] tthint: call stack

### PR DESCRIPTION
**Pull request by dfrg**
Fairly simple implementation of a stack to track nested function calls. Based on #791 which must land first.

### CODE REVIEW

File: skrifa/src/outline/glyf/hint/call_stack.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    pub return_pc: usize,
+    pub current_count: u32,
+    pub definition: Definition,
+}
+
+/// Tracker for nested active function or instruction calls.
+#[derive(Default)]
+pub struct CallStack {
+    records: [CallRecord; MAX_DEPTH],
+    top: usize,
```
Comment (rsheeter) [inline skrifa/src/outline/glyf/hint/call_stack.rs:25]: imho the behavior of this would be more intuitive if it was called "len" which more directly implies 0..len (non-inclusive) are the valid indices

File: skrifa/src/outline/glyf/hint/call_stack.rs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        Ok(record)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn stack_ops() {
```
Comment (rsheeter) [inline skrifa/src/outline/glyf/hint/call_stack.rs:67]: I suggest use of MAX_DEPTH not hard-coded 32 throughout.

I suggest this is several tests in one, I'd have expected to see a test that CallStack::default().pop() is underflow, a separate test for overflow, yet another test to fill/empty, etc.

**Review by rsheeter — APPROVED** [review]

File: skrifa/src/outline/glyf/hint/call_stack.rs (hunk 1 above)
Comment (dfrg) [reply skrifa/src/outline/glyf/hint/call_stack.rs]: Done

File: skrifa/src/outline/glyf/hint/call_stack.rs (hunk 2 above)
Comment (dfrg) [reply skrifa/src/outline/glyf/hint/call_stack.rs]: Oops, yes, changed to MAX_DEPTH. Separated the test into separate overflow/underflow/push&pop as suggested.
